### ISSUE TITLE
Component Edit Mode does not disable Entity Inspector interaction

### ISSUE THREAD

**Issue by LB-BartoszDudzinski** [o3de/o3de#5239]
**Describe the bug**
Entity Inspector interaction is not disabled after entering Component Edit Mode. The issue may lead to a crash when Component Edit Mode for a White Box component is active and a conflicting component is added and enabled.

**Steps to reproduce**
1. Create an entity with a White Box component.
2. Enter Component Edit Mode.
3. Add an Actor component and enable it.
4. Undo and then redo the last action.

**Expected behavior**
Entity Inspector interaction is disabled after entering Component Edit Mode.

**Actual behavior**
All of Entity Inspector interactions are available.

**Call Stack Trace:**
```
25) 00007FFA432DC347 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFA458F51B0 (ntdll) : memset
23) 00007FFA458DC766 (ntdll) : _C_specific_handler
22) 00007FFA458F20CF (ntdll) : _chkstk
21) 00007FFA458A1454 (ntdll) : RtlRaiseException
20) 00007FFA458F0BFE (ntdll) : KiUserExceptionDispatcher
19) 00007FF9BED8241A (WhiteBox.Editor) : DetachEnvironment
18) 00007FF9BED8F6F1 (WhiteBox.Editor) : DetachEnvironment
17) 00007FF9BED67B57 (WhiteBox.Editor) : DetachEnvironment
16) 00007FF9FC6B9F9C (EditorLib) : ModuleInitISystem
15) 00007FF9FC6D4D9B (EditorLib) : ModuleInitISystem
14) 00007FF9FC6D526B (EditorLib) : ModuleInitISystem
13) 00007FF9FCD3FC60 (EditorLib) : ModuleInitISystem
12) 00007FF9FCD25E2D (EditorLib) : ModuleInitISystem
11) 00007FF9FCD0936C (EditorLib) : ModuleInitISystem
10) 00007FF9FBC31556 (EditorLib) : QtViewport::wheelEvent
 9) 00007FF9FBC79CB7 (EditorLib) : EditorViewportWidget::RenderAll
 8) 00007FF9FBC73FDF (EditorLib) : EditorViewportWidget::OnBeginPrepareRender
 7) 00007FF9EF2E10E7 (Atom_RPI.Editor) : UninitializeDynamicModule
 6) 00007FF9EF2F36D6 (Atom_RPI.Editor) : UninitializeDynamicModule
 5) 00007FF9EF2F6ABB (Atom_RPI.Editor) : UninitializeDynamicModule
 4) 00007FF9FC463537 (EditorLib) : ModuleInitISystem
 3) 00007FF9FC4A0E17 (EditorLib) : ModuleInitISystem
 2) 00007FF9FC00B305 (EditorLib) : CCryEditApp::IdleProcessing
 1) 00007FF9FC0107E4 (EditorLib) : CCryEditApp::OnIdle
```

**Screenshots/Video**

https://user-images.githubusercontent.com/86952009/140061579-b86f72a8-4f3d-4d5b-b196-b74802ad1a09.mp4

**Found in Branch**
Stabilization/2110 (https://github.com/o3de/o3de/commit/5a0b6314e4da5446e8888bff7344203e73e09bd8)

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version 10
- CPU Intel I5-9600k
- GPU RTX 2060
- Memory 16GB

**Additional context**
[Editor.log](https://github.com/o3de/o3de/files/7467526/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7467527/error.log)
[dump.zip](https://github.com/o3de/o3de/files/7467528/dump.zip)

**Comment by hultonha** [comment]
Hi @LB-BartoszDudzinski, I'm just wondering is this is something that has only started failing recently? Or is it likely this has been the case for a while? It would just be helpful to know if any other recent changes caused this or not. Thanks!

**Comment by LB-BartoszDudzinski** [comment]
Hello @hultonha! I am not entirely sure, as today was the first time I've noticed this issue. My guess would be that maybe the prefab edit mode system which was introduced not so long ago. It has a focus mode similar to the Edit mode.

**Comment by jonawals** [comment]
Fixed here: https://github.com/o3de/o3de/pull/5486.

**Comment by LB-BartoszDudzinski** [comment]
Verified as fixed in Development (3791149) and Stabilization_2110 (7031147) branches.
